### PR TITLE
feat: add checksum algorithm to runtime plugin

### PR DESCRIPTION
**Pull request by AndrewFossAWS**
## Description of changes
* Add checksum algorithm to runtime plugin

## Scope

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/ClientRuntime/Config/DefaultClientConfiguration.swift [hunk 1]
```diff
@@ -29,5 +29,8 @@ public protocol DefaultClientConfiguration: ClientConfiguration {
     /// A token generator to ensure idempotency of requests.
     var idempotencyTokenGenerator: IdempotencyTokenGenerator { get set }
 
-    /// TODO(plugins): Add Checksum, Traceprobes, etc.
+    /// A checksum algorithm used for requests.
+    var checksumAlgorithm: ChecksumAlgorithm? { get set }
```
Comment (syall) [inline Sources/ClientRuntime/Config/DefaultClientConfiguration.swift:33]: Just double checking, do we need to support multiple checksum algorithms on a client configuration at once? e.g. [like smithy-typescript](https://github.com/smithy-lang/smithy-typescript/blob/e03a10ac1f34ac52e0b03509c13499e459bde3bc/packages/smithy-client/src/extensions/checksum.ts#L9-L15)

If not, then LGTM

**Review by syall — APPROVED** [review]

File: Sources/ClientRuntime/Networking/ChecksumAlgorithm.swift [hunk 2]
```diff
@@ -40,7 +40,7 @@ public enum ChecksumAlgorithm {
         return hashFunctions
     }
 
-    func toString() -> String {
+    public func toString() -> String {
```
Comment (dayaffe) [inline Sources/ClientRuntime/Networking/ChecksumAlgorithm.swift:43]: Clarification: It looks like the config has ChecksumAlgorithm enum and then the specified value is converted to a string to pass into FlexibleChecksumRequestMiddleware, is that correct?

**Review by dayaffe — COMMENTED** [review]
some small comments but overall LGTM (pending swiftlint fix)

File: Sources/ClientRuntime/Config/DefaultClientConfiguration.swift (hunk 1 above)
Comment (dayaffe) [reply Sources/ClientRuntime/Config/DefaultClientConfiguration.swift:33]: Question: Should the plugin use ChecksumAlgorithm enum (all possible supported checksum algorithms) or specifically the ones the particular client supports by looking in XClientTypes?

**Pull request closed without merging**